### PR TITLE
Propose Contributing documentation/policy

### PR DESCRIPTION
**Pull request by margau**

### CODE REVIEW

**Review by n0emis — APPROVED** [review]

**Review by leona-ya — APPROVED** [review]

File: dev/contr.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Contributing
+
+All contributions must be made via a Pull Request to the main GitHub-Repo.
+
+For a pull request to get merged, the following conditions must be fulfilled:
+
+- all applicable pipelines must succeed
+- at least two approvals must be made for the patch to be merged after 24 hours, or at least four approvals must be made to merge the patch immediately
+- the PR must neither be controversial or blocked (indicated by tags)
```
Comment (e1mo) [inline dev/contr.md:9]: I would also add a notice, that changes in the, database scheme and especially in areas that were disputed before, also require 4 approvals. What do you think?

Comment (adnidor) [reply dev/contr.md:9]: I think that should be covered by the controversial tag

**Review by adnidor — APPROVED** [review]

File: dev/contr.md (hunk 1 above)
Comment (margau) [reply dev/contr.md:9]: Maybe we should add a "fundamental"-state, requiring a deeper discussion and 4 approvals?

File: dev/contr.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+**Fundamental changes, like changes in the database scheme or other disputed areas, require at least four reviews. Those changes should be discussed in a developer meeting. A "Fundamental"-Tag should indicate those changes.**
```
Comment (e1mo) [inline dev/contr.md]:
```suggestion
**Fundamental changes, like changes in the database scheme or other disputed areas, require at least four reviews. Those changes should be discussed in a development meeting. A "Fundamental"-Tag should indicate those changes.**
```

File: dev/contr.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Contributing
+
+All contributions must be made via a Pull Request to the main GitHub-Repo.
+
+For a pull request to get merged, the following conditions must be fulfilled:
+
+- all applicable pipelines must succeed
+- at least two approvals must be made for the patch to be merged after 24 hours, or at least four approvals must be made to merge the patch immediately
```
Comment (e1mo) [inline dev/contr.md]:
```suggestion
- at least two approvals must are needed for the patch to be merged after 24 hours, or at least four approvals  to merge the patch immediately
```

File: dev/contr.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Contributing
+
+All contributions must be made via a Pull Request to the main GitHub-Repo.
```
Comment (e1mo) [inline dev/contr.md]:
```suggestion
All contributions must be made via a Pull Request to the main GitHub-Repository.
```

File: dev/contr.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+- at least two approvals must be made for the patch to be merged after 24 hours, or at least four approvals must be made to merge the patch immediately
+- the PR must neither be controversial or blocked (indicated by tags)
+
+**Fundamental changes, like changes in the database scheme or other disputed areas, require at least four reviews. Those changes should be discussed in a developer meeting. A "Fundamental"-Tag should indicate those changes.**
+
+The Documentation may be also expanded using pull requests. For the documentation repository, two approvals are enough for immediate merge.
+
+Usually, commits from a single PR are squashed and merged.
+
+PRs need to have a meaningful name, and, if applicable, a description.
```
Comment (e1mo) [inline dev/contr.md]: I would condense this into a listing like so:

```suggestion
The Documentation may be also expanded using pull requests (PRs). Pull requests need need to follow a few guidelines:

- A PR may be merged immediately with a minimum of only two approvals.
- Multiple commits must be squashed before or during merging, exceptions confirm the rule.
- PRs require a meaningful title and, if needed to explain the changes or the rational behind them, a proper description.
```

Comment (e1mo) [reply dev/contr.md]: We could also, I would even prefer that, use the same set of default guidelines (e.g. Title, squashing, approvals, guidelines for fundamental or controversial changes) for the Code and the Documentation and only defining differences, for example that a Documentation PR may be merged immediately when two contributors approved of it.

**Review by e1mo — CHANGES_REQUESTED** [review]

File: dev/contr.md (hunk 5 above)
Comment (margau) [reply dev/contr.md]: Point two: Really a git stash? Not squash?
The other points are looking good, thanks!

Comment (e1mo) [reply dev/contr.md]: I keep messing those two up... Fixed

File: dev/contr.md [hunk 6]
```diff
@@ -1,17 +1,17 @@
 # Contributing
 
-All contributions must be made via a Pull Request to the main GitHub-Repo.
+All contributions must be made via a Pull Request to the main GitHub-Repository.
 
 For a pull request to get merged, the following conditions must be fulfilled:
 
 - all applicable pipelines must succeed
-- at least two approvals must be made for the patch to be merged after 24 hours, or at least four approvals must be made to merge the patch immediately
+- at least two approvals must are needed for the patch to be merged after 24 hours, or at least four approvals  to merge the patch immediately
```
Comment (adnidor) [inline dev/contr.md]:
```suggestion
- at least two approvals are needed for the patch to be merged after 24 hours, or at least four approvals  to merge the patch immediately
```

File: dev/contr.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+- all applicable pipelines must succeed
+- at least two approvals must are needed for the patch to be merged after 24 hours, or at least four approvals  to merge the patch immediately
+- the PR must neither be controversial or blocked (indicated by tags)
+
+**Fundamental changes, like changes in the database scheme or other disputed areas, require at least four reviews. Those changes should be discussed in a development meeting. A "Fundamental"-Tag should indicate those changes.**
+
+The Documentation may be also expanded using pull requests (PRs). Pull requests need need to follow a few guidelines:
+
+- A PR may be merged immediately with a minimum of only two approvals.
+- Multiple commits must be squashed before or during merging, exceptions confirm the rule.
```
Comment (adnidor) [inline dev/contr.md]:
```suggestion
- Multiple commits should be squashed before or during merging, except where it would impact comprehensibility.
```

**Review by adnidor — CHANGES_REQUESTED** [review]

File: dev/contr.md (hunk 7 above)
Comment (margau) [reply dev/contr.md]: Applied suggestions, as it only affects details. Thanks!

**Review by clerie — APPROVED** [review]

**Review by adnidor — APPROVED** [review]

**Review by e1mo — APPROVED** [review]

**Review by leona-ya — APPROVED** [review]

**Review by n0emis — APPROVED** [review]

**Pull request merged**